modules/ardrivo/src/Servo.cxx

```diff
@@ -0,0 +1,56 @@
+/*
+ *  Servo.cxx
+ *  Copyright 2020 ItJustWorksTM
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+#include "Arduino.h"
+#include "BoardDataDef.hxx"
+#include "Servo.h"
+
+constexpr int servo_min(int8_t min) { return MIN_PULSE_WIDTH - min * 4; }
+constexpr int servo_max(int8_t max) { return MAX_PULSE_WIDTH - max * 4; }
+constexpr uint8_t TRIM_DURATION = 2;
+
+uint8_t Servo::attach(int pin) { return this->attach(pin, MIN_PULSE_WIDTH, MAX_PULSE_WIDTH); }
+
+uint8_t Servo::attach(int pin, int minWidth, int maxWidth) {
+    if (board_info->pins_caps[pin].pwm_able) {
```

Comment:
This is global state access; the first line of this function should be the function call `maybe_init();`
Same comment applies to `read`, `write`, and `writeMicroseconds`

Reply:
Didn't find function `maybe_init()`

Reply:
Right; forgot it was in a branch I haven't merged yet

Reply:
I will add that later then

Reply:
See #43